relay-sampling/src/config.rs

```diff
@@ -81,14 +83,33 @@ impl SamplingRule {
         self.condition.supported() && self.ty != RuleType::Unsupported
     }
 
-    /// Returns the sample rate if the rule is active.
-    pub fn sample_rate(&self, now: DateTime<Utc>) -> Option<SamplingValue> {
+    /// Returns `true` if rule is a reservoir rule.
+    pub fn is_reservoir(&self) -> bool {
+        matches!(&self.sampling_value, &SamplingValue::Reservoir { .. })
+    }
```

Comment:
This function is not referenced anywhere, and it's a rather specific function -- looks like it can be removed.

Reply:
nice catch yeah, stopped being used when I changed to no allocation removing counters